### PR TITLE
Rudr samples - bikesharing from Microsoft

### PR DESCRIPTION
**Pull request by sowsan**
Moving the sample to this repo as per the request from @hongchaodeng 

An original pull request can be found here: https://github.com/oam-dev/rudr/pull/506 

//
Added two new Rudr examples by using the existing Microsoft Bikesharing360 samples.

Multi-container app contains 3 APIs and a UI and single app is a one container app.

BikeSharing360 is a fictitious example of a smart bike-sharing system with 10,000 bikes distributed in 650 stations located throughout New York City and Seattle. Their vision is to provide a modern and personalized experience to riders and to run their business with intelligence.

This was a team effort from Microsoft OCP app dev CSAs.

### CODE REVIEW

**Comment by hongchaodeng** [comment]
lgtm

I'm leaving it open until after the holidays so that other people might take a look

**Comment by sowsan** [comment]
Sure, thanks 🙏

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: BikeSharing360_MC/Profile.Api/Manifest/bikesharing-profile-api-component.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        - name: ASPNETCORE_ENVIRONMENT
+          fromParam: aspnetcore_environment      
+      ports:
+        - protocol: TCP
+          containerPort: 80
+          name: http
+  parameters:
+    - default: Production
+      name: aspnetcore_environment
+      type: string
```
Comment (ryanzhang-oss) [inline BikeSharing360_MC/Profile.Api/Manifest/bikesharing-profile-api-component.yaml]: How come that I don't seem to see this component used in an application config?

**Comment by sowsan** [comment]
Thank you all.

**Comment by sowsan** [comment]
Thank you. What is the next step ?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by suhuruli** [comment]
LGTM. I will be merging this now. 

Sorry for the delay, I was OOF on holidays. Thanks for this contribution.

File: BikeSharing360_MC/.gitignore [hunk 2]
```diff
@@ -0,0 +1,353 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (suhuruli) [inline BikeSharing360_MC/.gitignore]: Do we need such a verbose gitignore for this?

Comment (sowsan) [reply BikeSharing360_MC/.gitignore]: Not really. What is your suggestion? Do we need to remove .gitignore? or just that entry.

**Comment by sowsan** [comment]
> Some changes requested:
> 
> * Is it possible to move the other sample into a folder? Given that this repo will now have two samples, we should have a README for each sample in it's folder along with a landing README for the repo which has a Table of Contents describing each sample. Thoughts?

yes, I put the same comment on other pull review. So so the root folder has a folder called samples and with in it app, bikesharing_sc, bikesharing_mc apps with its own readme. A general readme file in the root folder?

File: BikeSharing360_MC/.gitignore (hunk 2 above)
Comment (suhuruli) [reply BikeSharing360_MC/.gitignore]: It's a nit but the .gitignore was just pretty big and it looked like some entries not necessary (not just this one).

**Comment by suhuruli** [comment]
> > Some changes requested:
> > 
> > * Is it possible to move the other sample into a folder? Given that this repo will now have two samples, we should have a README for each sample in it's folder along with a landing README for the repo which has a Table of Contents describing each sample. Thoughts?
> 
> yes, I put the same comment on other pull review. So the root folder will have a folder called samples and with in it app, bikesharing_sc, bikesharing_mc apps with its own readme. A general readme file in the root folder?

OK. I was suggesting that we move hte other sample (the service-tracker) into a folder as well but I can do that once we merge this!

File: BikeSharing360_MC/.gitignore (hunk 2 above)
Comment (sowsan) [reply BikeSharing360_MC/.gitignore]: Got it, yeh it was added from vs/vs code. We haven't added anything extra. So, better would be trim it down and push the changes?

**Comment by sowsan** [comment]
> > > Some changes requested:
> > > 
> > > * Is it possible to move the other sample into a folder? Given that this repo will now have two samples, we should have a README for each sample in it's folder along with a landing README for the repo which has a Table of Contents describing each sample. Thoughts?
> > 
> > 
> > yes, I put the same comment on other pull review. So the root folder will have a folder called samples and with in it app, bikesharing_sc, bikesharing_mc apps with its own readme. A general readme file in the root folder?
> 
> OK. I was suggesting that we move hte other sample (the service-tracker) into a folder as well but I can do that once we merge this!

Got it. Thanks!

File: BikeSharing360_MC/.gitignore (hunk 2 above)
Comment (suhuruli) [reply BikeSharing360_MC/.gitignore]: If you can, that would be great!

Comment (sowsan) [reply BikeSharing360_MC/.gitignore]: ok, one more clarification, we used the existing samples from here https://github.com/microsoft/BikeSharing360_MultiContainer/ where I believe its the same .gitignore content. Is there a way that we can flush it to a default/minmal one? (havent done that before)

Comment (suhuruli) [reply BikeSharing360_MC/.gitignore]: I don't think there is any easy way. We can just keep this :)

**Pull request merged**